addOns/jwt/src/main/java/org/zaproxy/zap/extension/jwt/utils/JWTUtils.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.jwt.utils;
+
+import static org.zaproxy.zap.extension.jwt.utils.JWTConstants.BASE64_PADDING_CHARACTER_REGEX;
+import static org.zaproxy.zap.extension.jwt.utils.JWTConstants.BEARER_TOKEN_KEY;
+import static org.zaproxy.zap.extension.jwt.utils.JWTConstants.BEARER_TOKEN_REGEX;
+import static org.zaproxy.zap.extension.jwt.utils.JWTConstants.JWT_TOKEN_REGEX_PATTERN;
+
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import org.zaproxy.zap.extension.jwt.exception.JWTException;
+
+/**
+ * Contains Utility methods for handling various operations on JWT Token.
+ *
+ * @author KSASAN preetkaran20@gmail.com
+ * @since TODO add version
+ */
+public class JWTUtils {
+
+    /**
+     * Converts string to bytes. This method assumes that token is in UTF-8 charset which is as per
+     * the JWT specifications.
+     *
+     * @param token
+     * @return
+     */
+    public static byte[] getBytes(String token) {
+        return token.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Converts bytes to String. This method assumes that bytes provides are as per UTF-8 charset
+     *
+     * @param tokenBytes
+     * @return
+     */
+    public static String getString(byte[] tokenBytes) {
+        return new String(tokenBytes, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * we are using <a href="https://en.wikipedia.org/wiki/Base64#URL_applications">base64 Url Safe
+     * encoding</a>. because of JWT specifications <br>
+     * Also we are removing the padding as per <a
+     * href="https://www.rfc-editor.org/rfc/rfc7515.txt">RFC 7515</a> padding is not there in JWT.
+     *
+     * @param token
+     * @return
+     */
+    public static String getBase64UrlSafeWithoutPaddingEncodedString(String token) {
+        return JWTUtils.getBase64UrlSafeWithoutPaddingEncodedString(getBytes(token));
+    }
+
+    /**
+     * we are using <a href="https://en.wikipedia.org/wiki/Base64#URL_applications">base64 Url Safe
+     * encoding</a>. because of JWT specifications <br>
```

Comment:
```suggestion
     * Using <a href="https://en.wikipedia.org/wiki/Base64#URL_applications">Base64 URL Safe
     * encoding</a>. because of JWT specifications.<br>
```